packages/mdc-slider/component.ts

```diff
@@ -117,7 +117,13 @@ export class MDCSlider extends MDCComponent<MDCSliderFoundation> {
         this.thumbContainer_.style.setProperty(propertyName, value);
       },
       setTrackStyleProperty: (propertyName, value) => this.track_.style.setProperty(propertyName, value),
-      setMarkerValue: (value) => this.pinValueMarker_.innerText = value.toLocaleString(),
+      setMarkerValue: (value) => {
+        if (value === 0) {
+          this.pinValueMarker_.innerText = '0';
```

Comment:
I'm not sure if this is the appropriate solution. The problem is that setMarkerValue is being provided `-0` by the foundation, and it shouldn't be doing that.

The root problem is the [calculation of `pctComplete`](https://github.com/material-components/material-components-web/blob/master/packages/mdc-slider/foundation.ts#L330), which can be negative if the pageX is negative.

I think a better solution might be to solve the problem there. It's not possible for the slider to be "negative percentage" filled.

```diff
     let pctComplete = xPos / this.rect_.width;
+    if (pctComplete < 0) {
+      pctComplete = 0;
+    }
```

By fixing it in the foundation layer, we won't have to fix similar `-0` bugs in other implementations, such as MWC.

Can you make that change and resolve the conflicting file?

Reply:
Thanks for your reply, updated pull request with resolved conflicting file.

Tried your variant but looks like the bug is still there: the test is failing with `-0` slider value.
Here is the page similar to the one in the test: https://codepen.io/Tester456/pen/povOvEJ

Reply:
Ok, I understand. It's possible that "0" on the slider isn't 0%.

I still think that a foundation-layer solution is the appropriate approach. The component shouldn't have to worry about what its getting from the foundation or need to do any data massaging.

You could take advantage of JS's `||` pipe and the fact that both `0` and `-0` evaluate as falsy to coerce `-0` cases to `0`. [setValue_()](https://github.com/material-components/material-components-web/blob/master/packages/mdc-slider/foundation.ts#L435) would be an appropriate place to fix it.

```diff
+    value = value || 0; // coerce -0 to 0
     this.value_ = value;
     this.adapter_.setAttribute(strings.ARIA_VALUENOW, String(this.value_));
```

Reply:
Yes, I agree with you, that it should be fixed foundation, I just wasn't able to find the place where is should be fixed. Looks like it's working fine now.
Thanks a lot for the fix!